drivers/vl53l1x/vl53l1x.c

```diff
@@ -0,0 +1,1316 @@
+    EXEC_RET(_write_byte(dev, VL53L1X_SD_CONFIG__QUANTIFIER, 2));
+
+    /*
+     * from VL53L1_preset_mode_timed_ranging_*
+     * GPH is 0 after reset, but writing GPH0 and GPH1 above seem to set GPH
+     * to 1, and things don't seem to work if we don't set GPH back to 0 (which
+     * the API does here).
+     */
+    EXEC_RET(_write_byte(dev, VL53L1X_SYSTEM__GROUPED_PARAMETER_HOLD, 0x00));
+    EXEC_RET(_write_byte(dev, VL53L1X_SYSTEM__SEED_CONFIG, 1));
+
+    /* from VL53L1_config_low_power_auto_mode */
+    EXEC_RET(_write_byte(dev, VL53L1X_SYSTEM__SEQUENCE_CONFIG, 0x8B));
+    EXEC_RET(_write_word(dev, VL53L1X_DSS_CONFIG__MANUAL_EFFECTIVE_SPADS_SELECT, 200 << 8));
+    EXEC_RET(_write_byte(dev, VL53L1X_DSS_CONFIG__ROI_MODE_CONTROL, 2));
+
+    /* default to long range, 50 ms timing budget */
+    EXEC_RET(vl53l1x_set_distance_mode(dev, VL53L1X_DIST_LONG));
+    EXEC_RET(vl53l1x_set_timing_budget(dev, 50000));
+
+    /*
+     * the API triggers this change in VL53L1_init_and_start_range() once a
+     * measurement is started; assumes MM1 and MM2 are disabled
+     */
+    uint16_t word;
+    EXEC_RET(_read_word(dev, VL53L1X_MM_CONFIG__OUTER_OFFSET_MM, &word));
+    word *= 4;
+    EXEC_RET(_write_word(dev, VL53L1X_ALGO__PART_TO_PART_RANGE_OFFSET_MM, word));
+
+    /* Copyrighted POLOLU / ST API code end */
+
+#endif /* IS_USED(MODULE_VL53L1X_ST_API) */
+
+    /* set measurement parameters */
+    EXEC_RET(vl53l1x_set_distance_mode(dev, dev->params.mode));
+    EXEC_RET(vl53l1x_set_timing_budget(dev, dev->params.budget));
+    EXEC_RET(vl53l1x_set_measurement_period(dev, dev->params.period));
+
+#endif /* IS_USED(MODULE_VL53L1X_BASIC) */
+
+    /* start measurement */
+    EXEC_RET(_start(dev));
+
+    return VL53L1X_OK;
+}
+
+#define VL53L1X_BOOT_TIMEOUT    10      /* 10 ms */
+
+static int _reset(vl53l1x_t *dev)
+{
+#if IS_USED(MODULE_VL53L1X_ST_API)
+
+    /* write reset impuls for 150 us */
+    VL53L1_software_reset(&dev->dev);
+
+    /* give the sensor some time to boot */
+    ztimer_sleep(ZTIMER_MSEC, 4);
+
+    /* wait until the sensor has been booted */
+    EXEC_RET(VL53L1_WaitDeviceBooted(&dev->dev));
+
+    /* set VDDIO */
+    EXEC_RET(VL53L1_UpdateByte(&dev->dev,
+                               VL53L1X_PAD_I2C_HV__EXTSUP_CONFIG,
+                               0xfe, dev->params.vddio_2v8 ? 1 : 0));
+
+#else /* IS_USED(MODULE_VL53L1X_ST_API) */
+
+    /* write reset impuls of at least 100 us, we use 1 ms */
+    _write_byte(dev, VL53L1X_SOFT_RESET, 0);
+    ztimer_sleep(ZTIMER_MSEC, 1);
+    _write_byte(dev, VL53L1X_SOFT_RESET, 1);
+
+    /* wait until the sensor has been booted in 1 ms steps*/
+    unsigned timeout = VL53L1X_BOOT_TIMEOUT;
+    uint8_t res;
+    while (timeout--) {
+        if ((_read_byte(dev, VL53L1X_FIRMWARE__SYSTEM_STATUS, &res) == VL53L1X_OK) &&
+            (res & 0x01) == 1) {
+            break;
+        }
+        ztimer_sleep(ZTIMER_MSEC, 1);
+    }
+    if (timeout == 0) {
```

Comment:
```suggestion
    if (timeout == UINT_MAX) {
```

The postdecrement will still kick in, see:

```
~ $ cat test.c
#include <stdio.h>

int main(void) {
    unsigned foo = 3;
    while (foo--) { }
    printf("foo = %u\n", foo);

    return 0;
}
~ $ gcc -o test test.c
~ $ ./test
foo = 4294967295
```

But IMO it would be more readable to just go for

```C
while (timeout) {
    timeout--; 
    ...
```

instead.

Reply:
> The postdecrement will still kick in, see:

Yeah of course. Good catch. Fixed.